tests/transforms/test_compile.py

```diff
@@ -121,6 +121,147 @@ def test_compile_mcm(self):
+            qml.Hadamard(wires=1)
+            qml.Hadamard(wires=1)
+            return qml.expval(qml.Z(0))
+
+        assert np.allclose(circuit(0.5), 1.0)
+
+    def test_transform_multiplication_creates_pipeline(self):
+        """Test that transform * n creates a CompilePipeline."""
+        pipeline = qml.transforms.merge_rotations * 2
+        assert isinstance(pipeline, qml.CompilePipeline)
+        assert len(pipeline) == 2
+
+    def test_transform_addition_creates_pipeline(self):
+        """Test that transform + transform creates a CompilePipeline."""
+        pipeline = qml.transforms.cancel_inverses + qml.transforms.merge_rotations
+        assert isinstance(pipeline, qml.CompilePipeline)
+        assert len(pipeline) == 2
+
+    def test_in_place_addition(self):
+        """Test dynamically building CompilePipeline with +=."""
+        pipeline = qml.CompilePipeline()
+        pipeline += qml.transforms.cancel_inverses
+        pipeline += qml.transforms.merge_rotations
+
+        assert len(pipeline) == 2
+        assert qml.transforms.cancel_inverses in pipeline
+        assert qml.transforms.merge_rotations in pipeline
+
+    def test_nested_pipelines(self):
+        """Test CompilePipeline containing another CompilePipeline."""
+        inner = qml.CompilePipeline(qml.transforms.merge_rotations)
+        outer = qml.CompilePipeline(qml.transforms.cancel_inverses, inner)
+
+        assert len(outer) == 2
+
+    def test_pipeline_multiplication(self):
+        """Test that pipeline * n repeats the pipeline."""
+        pipeline = qml.CompilePipeline(qml.transforms.merge_rotations) * 3
+        assert len(pipeline) == 3
+
+    def test_empty_pipeline(self):
+        """Test that an empty CompilePipeline preserves original operations."""
+        dev = qml.device("default.qubit", wires=1)
+        pipeline = qml.CompilePipeline()
+
+        @qml.compile(pipeline=pipeline)
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.1, wires=0)
+            qml.RX(0.2, wires=0)
+            return qml.expval(qml.Z(0))
+
+        tape = qml.workflow.construct_tape(circuit)()
+        assert len([op for op in tape.operations if op.name == "RX"]) == 2
+
+    def test_bound_transforms(self):
+        """Test CompilePipeline with bound transforms (transforms with arguments)."""
+        bound_merge = qml.transforms.merge_rotations(atol=1e-2)
+        pipeline = qml.CompilePipeline(bound_merge)
+
+        assert len(pipeline) == 1
+
+    def test_remove_method(self):
+        """Test CompilePipeline remove() method."""
+        pipeline = qml.CompilePipeline(
+            qml.transforms.merge_rotations,
+            qml.transforms.cancel_inverses,
+        )
+        pipeline.remove(qml.transforms.cancel_inverses)
+
+        assert len(pipeline) == 1
+        assert qml.transforms.cancel_inverses not in pipeline
+        assert qml.transforms.merge_rotations in pipeline
+
+    def test_num_passes_with_pipeline(self):
```

Comment:
```suggestion
    def test_num_passes_with_pipeline(self):
```
This one also feels redundant since CompilePipeline already tested